internal/cmd/run.go

```diff
@@ -221,8 +221,10 @@ func (c *cmdRun) run(cmd *cobra.Command, args []string) (err error) {
 		}
 
 		switch summaryMode {
-		// TODO: Remove this code block once we stop supporting the legacy summary, and just leave the default.
+		// TODO(@joanlopez): remove by k6 v2.0, once we completely drop the support for --summary-mode=legacy.
 		case summary.ModeLegacy:
+			logger.Warn(`The --summary-mode=legacy has been deprecated, and will be removed by k6 v2.0. ` +
+				`Please, migrate to either "compact" or "full" as soon as possible.`)
```

Comment:
I doubt many people use the env variable for this (or the legacy value at all)  but technically this can be configured through the env variable as well.

Reply:
Right! So, would you suggest any more generic message? Any specific preference? cc/ @oleiade